pageserver/src/object_repository.rs

```diff
@@ -712,6 +753,17 @@ impl ObjectTimeline {
                         }
                         if last_version {
                             last_version = false;
+                            for timeline in descendant_timelines.keys() {
+                                let descendant_key = ObjectKey {
+                                    timeline: *timeline,
+                                    buf_tag: key.buf_tag,
+                                };
+                                if self.obj_store.oldest_version(&descendant_key)?.is_none() {
+                                    // No versions of this relation in descendant timeline:
```

Comment:
is it possible that the relation was dropped in the descendent timeline, and the dropped relation was already garbage collected away? In other words, we won't find any trace of the object in a descendent timeline not because it was not modified there, but because it was dropped there.

Reply:
Yeh. It is really possible. And I do not know how to distinguish this two situations.
But IMHO it is not the worst case. Usually fork modifies only some small fraction of objects of ancestor branch. But just presence of this branch will prevent GC in main branch from cleanup of most of objects which were not updated in  descendant branch (and so there are no versions in this timeline).

Actually it once again demonstrates that snapshots/backups should be treated indifferent way from forks/branches. Fork is used to be short living object and delaying GC for lifetime of fork should not critical. Most of snapshots/backups will never be used but them can live quote long time (years). So snapshot should not pin GC. It means that it needs to materialize all pages.